### PR TITLE
Insulin model delay

### PR DESCRIPTION
**Pull request by dm61**
Adds delay to the insulin model presets, reverts hard coded delay and subtraction of delay in DoseMath. This is a companion PR to LoopKit PR https://github.com/LoopKit/LoopKit/pull/315. Upon review, and any further updates, the two PR's should be merged concurrently. This Loop PR and the LoopKit PR 315 serve the same main purpose of fixing the dose math as https://github.com/LoopKit/Loop/pull/1238, which should be closed once these two PR's are merged.

### CODE REVIEW

File: LoopCore/Insulin/ExponentialInsulinModelPreset.swift [hunk 1]
```diff
@@ -38,9 +38,20 @@ extension ExponentialInsulinModelPreset {
             return .minutes(55)
         }
     }
+    
+    var effectDelay: TimeInterval {
```
Comment (sraka1) [inline LoopCore/Insulin/ExponentialInsulinModelPreset.swift]: @dm61 ideally, this should be user configurable (due to YDMV) IMHO. In many cases 15 or even 20 minutes might be more adequate (without the user needing to make a code change).

Comment (dm61) [reply LoopCore/Insulin/ExponentialInsulinModelPreset.swift]: Thanks for the comment. The main purpose of this PR is to address a slight inconsistency in dosing versus prediction algorithms, as discussed in  https://github.com/LoopKit/Loop/pull/1238. UI-related decisions and design are separate issues, beyond the scope of this particular PR.

**Comment by Trpl7ca** [comment]
is it possible to merge this into automatic-bolus branch?

**Comment by ps2** [comment]
Thanks @dm61. Looks great!

**Comment by sarisarisari** [comment]
Can I please clarify that this means that the algorhythm will now assume that the insulin doesn't begin it's function until 10 min after it's been programmed? Where previously the algorhythm assumed immediate action?

**Comment by ps2** [comment]
The main algorithm has always assumed a 10 minute delay, and you would see this in the forecast, IOB charts, etc.  The fix here is how dosing is calculated.  Loop iterates to see what size dose will correct any highs without bringing you below range. That calculation wasn't using the 10 minute delay, so doses could be smaller, depending on the timing of carbs and your current bg.

**Comment by sarisarisari** [comment]
Oh interesting.  Ok. Thank you for the clarification.

**Pull request merged**